codegenerator/cli/src/config_parsing/system_config.rs

```diff
@@ -45,10 +47,327 @@ pub enum Ecosystem {
     Fuel,
 }
 
+// Allows to get an env var with a lazy loading of .env file
+#[derive(Debug)]
+struct EnvState {
+    // Lazy loading of .env file
+    maybe_dotenv: Option<EnvMap>,
+    project_root: PathBuf,
+}
+
+impl EnvState {
+    fn new(project_root: &PathBuf) -> Self {
+        EnvState {
+            maybe_dotenv: None,
+            project_root: project_root.clone(),
+        }
+    }
+
+    fn var(&mut self, name: &str) -> Option<String> {
+        match std::env::var(name) {
+            Ok(val) => Some(val),
+            Err(_) => {
+                let result = match &self.maybe_dotenv {
+                    Some(env_map) => env_map.var(name),
+                    None => {
+                        match EnvLoader::with_path(self.project_root.join(".env"))
+                            .sequence(EnvSequence::InputOnly)
+                            .load()
+                        {
+                            Ok(env_map) => {
+                                self.maybe_dotenv = Some(env_map.clone());
+                                env_map.var(name)
+                            }
+                            Err(err) => {
+                                match err {
+                              dotenvy::Error::Io(_, _) => (),
+                              _ => println!("Warning: Failed loading .env file with unexpected error: {err}"),
+                          };
```

Comment:
Think the indentation is off here, maybe running `make format` will help.

Reply:
So the idea here is that if it fails to load the file because it's not there that is fine but otherwise the error gets reported?

Just checking the warning won't be printed in this case.

Reply:
Exactly